.github/workflows/build_and_deploy_ext2_img.yml

```diff
@@ -0,0 +1,133 @@
+      # We create our ext2 filesystem.
+      # We preallocate space for the ext2 image
+      - run: sudo fallocate -l $IMAGE_SIZE ${DEST}${IMAGE_NAME}
+      # Format to ext2 linux kernel revision 0
+      - run: sudo mkfs.ext2 -r 0 ${DEST}${IMAGE_NAME}
+      # We mount the ext2 image in order to modify it.
+      - run: | 
+          sudo mkdir -p /mnt/ext2_image
+          sudo mount -o loop -t ext2 ${DEST}/${IMAGE_NAME} /mnt/ext2_image
+      # We export our container its filesystem and unpack it into the mounted ext2 image.
+      - run: | 
+          docker export $(sudo docker ps -aq) | sudo tar -x -C /mnt/ext2_image
+          sudo umount /mnt/ext2_image
+  
+      # Result is an ext2 image for webvm.
+      - name: Upload GitHub Pages artifact
+        uses: actions/upload-pages-artifact@v1.0.7
+        with:
+          # Path of the directory containing the static assets.
+          path: ${{ env.DEST }} # default is _site/
+          # Duration after which artifact will expire in days.
+          retention-days: 90 # optional, default is 1
+
+      - name: Check if release_tag present in case of gh release.
+        if: ${{ github.event.inputs.GITHUB_RELEASE == 'true' }}
+        shell: bash
+        run: | # Check tag_name value. Set to default in case empty.
+          if [ "${{ github.event.inputs.GITHUB_RELEASE_TAG }}" == '' ]; then
+            echo "GITHUB_RELEASE_TAG=ext2_image" >> $GITHUB_ENV
+          else
+            echo "GITHUB_RELEASE_TAG=${{ github.event.inputs.GITHUB_RELEASE_TAG }}" >> $GITHUB_ENV
+          fi
+    
+      # To upload a github release.
+      - name: github release
+        if: ${{ github.event.inputs.GITHUB_RELEASE == 'true' }}
+        uses: softprops/action-gh-release@v0.1.15
+        with:
+          target_commitish: ${{ github.sha }} # Last commit on the GITHUB_REF branch or tag
+          tag_name: '${{ env.GITHUB_RELEASE_TAG }}' # Tag of the release.
+          fail_on_unmatched_files: 'true' # Fail in case of no matches with the file(s) glob(s).
+          files: | # Assets to upload as release.
+            ${{ env.DEST }}${{ env.IMAGE_NAME }}
+
+  deploy_to_pages: # Job that deploys the github-pages artifact to github-pages.
+    if: ${{ github.event.inputs.DEPLOY_TO_PAGES == 'true' }}
+    needs: build
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+
+    # Grant GITHUB_TOKEN the permissions required to make a Pages deployment
+    permissions:
+      pages: write      # to deploy to Pages
+      id-token: write   # to verify the deployment originates from an appropriate source
+
+    runs-on: ubuntu-latest
+    steps:
+      # Deployment to github pages
+      - name: Deploy GitHub Pages site
+        id: deployment
+        uses: actions/deploy-pages@v2.0.0
+      - name: Print final url
+        run: echo "::notice::URL to the image ${{ steps.deployment.outputs.page_url }}${{ github.event.inputs.IMAGE_NAME }}"
```

Comment:
Hmmm... we should deploy the webvm HTML page + JS scripts and asset too. The whole point is to make a fully executable version